docs/tutorial.md

```diff
@@ -0,0 +1,557 @@
+    ) : Presenter<InboxScreen.State> {
+      // ...
+      class Factory(private val emailRepository: EmailRepository) : Presenter.Factory {
+        override fun create(screen: Screen, navigator: Navigator, context: CircuitContext): Presenter<*>? {
+          return when (screen) {
+            InboxScreen -> return InboxPresenter(navigator, emailRepository)
+            else -> null
+          }
+        }
+      }
+    }
+    ```
+
+=== "Circuit instance"
+    ```kotlin hl_lines="3-4"
+    val circuit: Circuit =
+      Circuit.Builder()
+        .addPresenterFactory(InboxPresenter.Factory(emailRepository))
+        .addUi<InboxScreen, InboxScreen.State> { state, modifier -> Inbox(state, modifier) }
+        .addPresenterFactory(DetailPresenter.Factory(emailRepository))
+        .addUi<DetailScreen, DetailScreen.State> { state, modifier -> EmailDetail(state, modifier) }
+        .build()
+    ```
+
+Now that we have a `Navigator` in our inbox presenter, we can use it to navigate to the detail screen. First, we need to explore how events work in Circuit.
+
+## Events
+
+So far, we've covered _state_. State is produced by the presenter and consumed by the UI. That's only half of the UDF picture though! _Events_ are the inverse: they're produced by the UI and consumed by the presenter. Events are how you can trigger actions in your app, such as navigation. _This completes the circuit._
+
+Events in Circuit are a little unconventional in that Circuit doesn't provide structured APIs for pipelining events from the UI to presenters. Instead, we use an _event sink property_ pattern, where states contain a trailing `eventSink` function that accepts events and can be emitted to by the UI.
+
+This provides many benefits, see the [events](states-and-events.md) guide for more information.
+
+Let's add an event to our inbox screen for when the user clicks on an email.
+
+Events must implement `CircuitUiEvent` ([docs](https://slackhq.github.io/circuit/api/0.x/circuit-runtime/com.slack.circuit.runtime/-circuit-ui-event/index.html)) and are usually modeled as a `sealed interface` hierarchy, where each subtype is a different event type.
+
+```kotlin title="InboxScreen.kt" hl_lines="4 6-8"
+data object InboxScreen : Screen {
+  data class State(
+    val emails: List<Email>,
+    val eventSink: (Event) -> Unit
+  ) : CircuitUiState
+  sealed class Event : CircuitUiEvent {
+    data class EmailClicked(val emailId: String) : Event()
+  }
+}
+```
+
+Now that we have an event, let's emit it from our UI.
+
+```kotlin title="InboxScreen.kt" hl_lines="6-8 15"
+@Composable
+fun Inbox(state: InboxScreen.State, modifier: Modifier = Modifier) {
+  Scaffold(modifier = modifier, topBar = { TopAppBar(title = { Text("Inbox") }) }) { innerPadding ->
+    LazyColumn(modifier = Modifier.padding(innerPadding)) {
+      items(state.emails) { email ->
+        EmailItem(email) {
+          state.eventSink(InboxScreen.Event.EmailClicked(email.id))
+        }
```

Comment:
nit: Lets name the `onClick` to line up with compose expectations that a trailing lambda is a composable function. 
```suggestion
        EmailItem(
          email = email,
          onClick = { state.eventSink(InboxScreen.Event.EmailClicked(email.id)) },
        )
```

Reply:
Does compose expect this for non-composable function/classes? For example [BackHandler](https://developer.android.com/reference/kotlin/androidx/activity/compose/package-summary#BackHandler(kotlin.Boolean,kotlin.Function0))

Reply:
Just for ui elements/layouts components, [docs](https://github.com/androidx/androidx/blob/androidx-main/compose/docs/compose-component-api-guidelines.md#parameters-order)